nautobot_data_validation_engine/template_content.py

```diff
@@ -0,0 +1,40 @@
+"""Template content for nautobot_data_validation_engine."""
+from django.contrib.contenttypes.models import ContentType
+from django.urls import reverse
+from nautobot.extras.plugins import TemplateExtension
+from nautobot_data_validation_engine.models import ValidationResult
+
+
+def tab_factory(content_type_label):
+    """Generate a ValidationTab object for a given content type."""
+
+    class ValidationTab(TemplateExtension):  # pylint: disable=W0223
+        """Dynamically generated ValidationTab class."""
+
+        model = content_type_label
+
+        def detail_tabs(self):
+            app_label, model = self.model.split(".")
+            return (
+                [
+                    {
+                        "title": "Validations",
+                        "url": reverse(
+                            "plugins:nautobot_data_validation_engine:validationresults",
+                            kwargs={"id": self.context["object"].id, "model": self.model},
+                        ),
+                    }
+                ]
+                if ValidationResult.objects.filter(
+                    content_type__app_label=app_label, content_type__model=model
+                ).exists()
+                else []
```

Comment:
It is confusing to not see the validation tab when there are no validations. I think the tab should be there even if no validations have been defined for the model.

```suggestion
```

Reply:
I think we either remove this and have the Validation Tab show up for all content types, or we filter the content types to just the ones that show up in a `ValidationSet` class in `validations.py`.

Reply:
Honestly, I think we should have it as a setting. 

* Always
* When Model has a validation
* When object has a validation

Reply:
Added a plugin setting to do this

https://github.com/nautobot/nautobot-plugin-data-validation-engine/pull/50/files#diff-00cd5ada92023e3b46e37806fc66ac9cfb88b021a25a5b87239f4e7b072fd0f7R18-R29